--- src/IconView.cs.orig	2006-08-15 14:58:28.000000000 +1000
+++ src/IconView.cs	2006-08-15 16:35:35.000000000 +1000
@@ -1416,13 +1416,19 @@
 			FocusCell = collection.Count - 1; 
 			break;
 		case Gdk.Key.space:
+			if (FocusCell < 0 || FocusCell > collection.Count - 1) {
+				FocusCell = 0;
+				break;
+			}
 			ToggleCell (FocusCell);
 			break;
 		case Gdk.Key.Return:
 			if (DoubleClicked == null)
 				break;
-			if (FocusCell < 0 || FocusCell > collection.Count - 1)
+			if (FocusCell < 0 || FocusCell > collection.Count - 1) {
+				FocusCell = 0;
 				break;
+			}
 			DoubleClicked (this, FocusCell);
 			break;
 		default:	
@@ -1433,7 +1439,7 @@
 		FocusCell = Math.Max (FocusCell, 0);
 		FocusCell = Math.Min (FocusCell, collection.Count - 1);
 		
-		if (FocusCell == focus_old) {
+		if (FocusCell < 0 || FocusCell == focus_old) {
 			args.RetVal = false;
 			return;
 		}
