--- src/IconView.cs.orig	2006-09-01 12:13:13.000000000 +1000
+++ src/IconView.cs	2006-09-01 13:02:36.000000000 +1000
@@ -150,6 +150,8 @@
 	public int FocusCell {
 		set {
 			if (value != real_focus_cell) {
+				value = Math.Max (value, 0);
+				value = Math.Min (value, collection.Count - 1);
 				InvalidateCell (value);
 				InvalidateCell (real_focus_cell);
 				real_focus_cell = value;
@@ -164,9 +166,6 @@
 	// about.
 	private int click_count;
 
-	// The pixbuf we use when we can't load a thumbnail.
-	static Pixbuf error_pixbuf;
-
 	// Public events.
 	public delegate void DoubleClickedHandler (IconView view, int clicked_item);
 	public event DoubleClickedHandler DoubleClicked;
@@ -1421,8 +1420,6 @@
 		case Gdk.Key.Return:
 			if (DoubleClicked == null)
 				break;
-			if (FocusCell < 0 || FocusCell > collection.Count - 1)
-				break;
 			DoubleClicked (this, FocusCell);
 			break;
 		default:	
@@ -1430,9 +1427,6 @@
 			return;		
 		}
 		
-		FocusCell = Math.Max (FocusCell, 0);
-		FocusCell = Math.Min (FocusCell, collection.Count - 1);
-		
 		if (FocusCell == focus_old) {
 			args.RetVal = false;
 			return;
