--- src/InfoBox.cs.orig	2006-08-18 04:13:44.000000000 +1000
+++ src/InfoBox.cs	2006-08-18 04:14:44.000000000 +1000
@@ -270,8 +270,8 @@
 		date_label.Text = info.Date;
 #else
 		date_label.Text = String.Format ("{0}\n{1}",
-						 photo.Time.ToShortDateString (),
-						 photo.Time.ToShortTimeString ());
+						 photo.Time.ToLocalTime ().ToShortDateString (),
+						 photo.Time.ToLocalTime ().ToShortTimeString ());
 #endif
 		version_option_menu.Sensitive = true;
 		PhotoVersionMenu menu = new PhotoVersionMenu (photo);
