--- tools/crawl-rules/crawl-applications.orig	2006-06-21 18:44:42.000000000 +1000
+++ tools/crawl-rules/crawl-applications	2006-06-21 18:45:19.000000000 +1000
@@ -2,7 +2,7 @@
 
 CRAWL_INDEX_NAME="applications"
 
-CRAWL_PATHS="/usr/share/applications /usr/local/share/applications /opt/gnome/share/applications /opt/kde3/share/applications"
+CRAWL_PATHS="/usr/share/applications /usr/local/share/applications"
 
 CRAWL_RECURSIVE="yes"
 
--- tools/crawl-rules/crawl-documentation.orig	2006-06-21 18:45:35.000000000 +1000
+++ tools/crawl-rules/crawl-documentation	2006-06-21 18:46:38.000000000 +1000
@@ -2,7 +2,7 @@
 
 CRAWL_INDEX_NAME="documentation"
 
-CRAWL_PATHS="/usr/share/doc /usr/local/share/doc /opt/kde3/share/doc /opt/gnome/share/gnome/help /usr/share/gnome/help /opt/gnome/share/gtk-doc/html /usr/share/gtk-doc/html /usr/share/gnome/html"
+CRAWL_PATHS="/usr/share/doc /usr/local/share/doc /usr/share/gnome/help /usr/share/gtk-doc/html /usr/share/gnome/html"
 
 CRAWL_ALLOW_PATTERNS="*.xml,*.html,*.docbook"
 
--- tools/crawl-rules/crawl-portage.orig	1970-01-01 10:00:00.000000000 +1000
+++ tools/crawl-rules/crawl-portage	2006-06-21 18:48:18.000000000 +1000
@@ -0,0 +1,18 @@
+# Beagle Portage Crawl Setup
+
+source /etc/make.globals
+source /etc/make.conf
+
+CRAWL_ENABLED="yes"
+
+CRAWL_INDEX_NAME="portage"
+
+CRAWL_PATHS="${PORTDIR} ${PORTDIR_OVERLAY} /var/db/pkg"
+
+CRAWL_ALLOW_PATTERNS="*.ebuild,ChangeLog,metadata.xml"
+
+CRAWL_RECURSIVE="yes"
+
+CRAWL_DISABLE_FILTERING="no"
+
+CRAWL_CACHE_TEXT="no"
--- tools/Makefile.am.orig	2006-06-21 19:20:29.000000000 +1000
+++ tools/Makefile.am	2006-06-21 19:20:56.000000000 +1000
@@ -35,9 +35,9 @@
 	chmod +x $(CRAWL_WRAPPER)
 
 CRAWL_RULES = \
-	$(srcdir)/crawl-rules/crawl-windows		\
 	$(srcdir)/crawl-rules/crawl-applications	\
-	$(srcdir)/crawl-rules/crawl-documentation
+	$(srcdir)/crawl-rules/crawl-documentation	\
+	$(srcdir)/crawl-rules/crawl-portage
 
 crondir = $(sysconfdir)/cron.daily
 cron_SCRIPTS = $(CRAWL_WRAPPER)
