--- configure.in.orig	2006-11-22 00:06:25.000000000 +1100
+++ configure.in	2006-11-22 00:13:17.000000000 +1100
@@ -357,6 +357,21 @@
 
 dnl ----------------------------------------------
 
+dnl wv1
+
+enable_wv1=no
+if test "x$enable_gsf_sharp" = "xyes"; then
+	PKG_CHECK_MODULES(WV1, wv-1.0, enable_wv1=yes, enable_wv1=no)
+	AC_SUBST(WV1_LIBS)
+fi
+
+AM_CONDITIONAL(ENABLE_WV1, test "x$enable_wv1" = "xyes")
+if test "x$enable_wv1" = "xyes"; then
+	BEAGLE_DEFINES="$BEAGLE_DEFINES -define:ENABLE_WV1"
+fi
+
+dnl ----------------------------------------------
+
 dnl libexif: Needed to index jpeg metadata
 
 LIBEXIF_API_CHANGE=0.6.0
@@ -535,16 +550,6 @@
 
 dnl ----------------------------------------------
 
-dnl For the wv1 glue
-
-PKG_CHECK_MODULES(WV1, wv-1.0, enable_wv1=yes, enable_wv1=no)
-AM_CONDITIONAL(ENABLE_WV1, test "x$enable_wv1" = "xyes")
-if test "x$enable_wv1" = "xyes"; then
-	BEAGLE_DEFINES="$BEAGLE_DEFINES -define:ENABLE_WV1"
-fi
-AC_SUBST(WV1_LIBS)
-
-dnl ----------------------------------------------
 
 dnl KDE launcher/icon support
 
