--- tools/crawl-rules/crawl-applications.orig	2006-11-03 06:11:21.000000000 +1100
+++ tools/crawl-rules/crawl-applications	2006-11-03 06:11:51.000000000 +1100
@@ -2,7 +2,7 @@
 
 CRAWL_INDEX_NAME="applications"
 
-CRAWL_PATHS="/usr/share/applications /usr/local/share/applications /opt/gnome/share/applications /opt/kde3/share/applications"
+CRAWL_PATHS="/usr/share/applications /usr/local/share/applications"
 
 CRAWL_RECURSIVE="yes"
 
--- tools/crawl-rules/crawl-documentation.orig	2006-11-03 06:12:01.000000000 +1100
+++ tools/crawl-rules/crawl-documentation	2006-11-03 06:12:44.000000000 +1100
@@ -2,7 +2,7 @@
 
 CRAWL_INDEX_NAME="documentation"
 
-CRAWL_PATHS="/usr/share/doc /usr/local/share/doc /opt/kde3/share/doc /opt/gnome/share/gnome/help /usr/share/gnome/help /opt/gnome/share/gtk-doc/html /usr/share/gtk-doc/html /usr/share/gnome/html"
+CRAWL_PATHS="/usr/share/doc /usr/local/share/doc /usr/share/gnome/help /usr/share/gtk-doc/html /usr/share/gnome/html"
 
 CRAWL_ALLOW_PATTERNS="*.xml,*.html,*.docbook"
 
--- tools/crawl-rules/crawl-portage.orig	1970-01-01 10:00:00.000000000 +1000
+++ tools/crawl-rules/crawl-portage	2006-11-03 06:15:40.000000000 +1100
@@ -0,0 +1,20 @@
+# Beagle Portage Crawl Setup
+
+source /etc/make.globals
+source /etc/make.conf
+
+CRAWL_ENABLED="no"
+
+CRAWL_INDEX_NAME="portage"
+
+CRAWL_PATHS="${PORTDIR} ${PORTDIR_OVERLAY} /var/db/pkg"
+
+CRAWL_ALLOW_PATTERNS="*.ebuild,ChangeLog,metadata.xml"
+
+CRAWL_RECURSIVE="yes"
+
+CRAWL_DISABLE_FILTERING="no"
+
+CRAWL_DISABLE_DIRECTORIES="yes"
+
+CRAWL_CACHE_TEXT="no"
--- tools/Makefile.am.orig	2007-02-09 20:57:01.000000000 +1100
+++ tools/Makefile.am	2007-02-09 20:57:34.000000000 +1100
@@ -36,9 +36,9 @@
 crondir = $(sysconfdir)/cron.daily
 
 CRAWL_RULES = \
-	$(srcdir)/crawl-rules/crawl-windows		\
 	$(srcdir)/crawl-rules/crawl-applications	\
-	$(srcdir)/crawl-rules/crawl-documentation
+	$(srcdir)/crawl-rules/crawl-documentation	\
+	$(srcdir)/crawl-rules/crawl-portage
 
 # FIXME: Per-distribution specific crawl rules
 configdir = $(sysconfdir)/beagle
