--- /var/tmp/portage/JavaScript-0.55/work/JavaScript-0.55/Makefile.PL	2005-06-29 07:49:03.000000000 +1000
+++ Makefile.PL	2006-06-25 03:11:14.000000000 +1000
@@ -14,7 +14,7 @@
 
 if(-f "/etc/gentoo-release") {
 	# Gentoo puts libjs in a slightly hidden spot.
-	$inc = '-I/usr/lib/MozillaFirefox/include/js/ -I/usr/lib/MozillaFirefox/';
+	$inc = '-I/usr/lib/mozilla-firefox/include/js/ -I/usr/lib/mozilla-firefox/ -I/usr/include/js/';
 }
 
 #test for debian type include files...
