--- src/systray.py.orig	2006-09-04 10:47:07.000000000 +1000
+++ src/systray.py	2006-09-04 10:48:12.000000000 +1000
@@ -373,7 +373,7 @@
 
 	def on_clicked(self, widget, event):
 		self.on_tray_leave_notify_event(widget, None)
-		if event.button == 1: # Left click
+		if event.type == gtk.gdk.BUTTON_PRESS and event.button == 1: # Left click
 			self.on_left_click()
 		elif event.button == 2: # middle click
 			self.on_middle_click()
